### ISSUE TITLE
Instructions/Name: Parse destination tag

### ISSUE THREAD

**Issue by boldtrn** [graphhopper/graphhopper#1086]
A nice addition for instructions would be to use the [destination tag](https://wiki.openstreetmap.org/wiki/Key:destination). This could be especially handy for _link roads, where we could create instructions like: take exit, following signs to "Berlin". One example would be this [link](http://www.openstreetmap.org/way/154765839) that misses the name tag, but has a ref.

The question is, how to store the data. 

Probably the easiest solution would be to store this in the name field. This was discussed in the topic about adding different languages for the name tag (couldn't find the issue, sorry :)).

I could implement this, if we can decide on a way to store this.

**Comment by devemux86** [comment]
@boldtrn you probably mean #259.
(where I propose the process we use for multilingual Mapsforge maps in writer / readers)

**Comment by boldtrn** [comment]
@devemux86, yes exactly, thanks for linking it. I used the wrong search terms, I think :).

**Comment by karussell** [comment]
> if we can decide on a way to store this

This feels wrong. Let us first find a better way of storing this and if this turns out to be too complex we can still use the name string.

**Comment by boldtrn** [comment]
> This feels wrong. Let us first find a better way of storing this and if this turns out to be too complex we can still use the name string.

Ok, what do you propose?

Should we use something like NameIndex, but call it DestinationIndex? Or should we try to extend the NameIndex to store multiple values, like different languages and different purposes like destination? 

Creating a second index feels a bit like an overkill, but the implementation would be straight forward as we can reuse a lot of existing code from the NameIndex.

Creating a more flexible storing solution would be probably the best, but how would we do this?

**Comment by karussell** [comment]
> Creating a more flexible storing solution would be probably the best, but how would we do this?

We would need a more flexible key/value storage to store more complicated properties per edge.

**Comment by boldtrn** [comment]
> We would need a more flexible key/value storage to store more complicated properties per edge.

Yes I know, that's why I am asking. However, at this point I don't have a good proposal. Easy solutions like serializing key/value pairs as JSON come to my mind, but this might be too slow. If we only use the name for creating instructions, this might be still ok, but we shouldn't store anything that we want to use during the route calculation.

**Comment by Penegal** [comment]
That would be a great improvement, especially for complex junctions in which reading the destination signs can be quicker than looking for the good road following the instructions.
